src/webots/nodes/utils/WbNodeOperations.cpp

```diff
@@ -83,6 +83,8 @@ void WbNodeOperations::cleanup() {
 }
 
 WbNodeOperations::WbNodeOperations() : mNodesAreAboutToBeInserted(false), mSkipUpdates(false), mFromSupervisor(false) {
+  connect(this, &WbNodeOperations::changedFromSupervisor, WbProtoManager::instance(),
+          &WbProtoManager::setImportedFromSupervisor);
```

Comment:
Do you really need a signal/slot mechanism?
Does not work if you call directly `WbProtoManager::setImportedFromSupervisor`?

Reply:
It would indeed work, but is it not clearer this way? Both global variables should have the same value, so a signal/slot mechanism seems the most appropriate to reflect this property.

Reply:
I agree with @stefaniapedrazzi. Signal/slot should be used only when we cannot call a method directly. Using signal/slot usually complicates the code and the debugging, so we should avoid it wherever possible.

Reply:
Signal/slot make the codes much more complicated and are heavier to run.
If possible we should call the functions directly and avoid adding extra layers.

I don't see exactly why it should be cleaner to emit a signal instead of calling a function, to me they look exactly the same.
What would make the code cleaner would be to define a function that sets both `WbNodeOperations::mFromSupervisor` and calls `WbProtoManager::setImportedFromSupervisor` to ensure that they always have the same value.

Reply:
Yes, we just discussed with @omichel and we agreed on using a function that would set both variables at the same time.